### PR TITLE
`ReadOnlyProject` functions that fetch `DataFrame` are now concurrent.

### PR DESCRIPTION
**Pull request by kgodlewski**
Make functions in `ReadOnlyProject` that return `pandas.DataFrame` fetch data concurrently.
This speeds up data download by 20-70 percent, depending on the data layout.

### CODE REVIEW

File: src/neptune_fetcher/read_only_project.py [hunk 1]
```diff
@@ -688,7 +719,7 @@ def _stream_runs(
         columns=["sys/id"],
         limit=limit,
         sort_by=sort_by,
-        step_size=1000,
+        step_size=FETCH_RUNS_BATCH_SIZE,
```
Comment (PatrykGala) [inline src/neptune_fetcher/read_only_project.py:722]: `step_size=FETCH_RUNS_BATCH_SIZE,`?

File: src/neptune_fetcher/read_only_project.py [hunk 2]
```diff
@@ -495,33 +497,62 @@ def _fetch_df(
...
-            if len(all_run_ids) > limit:
-                raise ValueError(
-                    f"The number of runs returned exceeds the limit of {limit}. "
-                    "Please narrow down your query or provide a smaller 'limit' "
-                    "as an argument"
+        # Workers fetching attributes in parallel
+        futures = []
+
+        value_count = 0
+        with concurrent.futures.ThreadPoolExecutor(max_workers=MAX_WORKERS) as executor:
```
Comment (PatrykGala) [inline src/neptune_fetcher/read_only_project.py:504]: Please check the number of connections in http client.

Comment (PatrykGala) [reply src/neptune_fetcher/read_only_project.py:504]: In near future we will use asyncio

**Review by PatrykGala — CHANGES_REQUESTED** [review]
LGTM

File: src/neptune_fetcher/read_only_project.py (hunk 1 above)
Comment (kgodlewski) [reply src/neptune_fetcher/read_only_project.py:722]: It was supposed to be this way, as `step_size` controls the page size retrieved from the backend. For some reason it was hardcoded.

**Review by michalsosn — APPROVED** [review]

**Review by PatrykGala — APPROVED** [review]

**Pull request merged**